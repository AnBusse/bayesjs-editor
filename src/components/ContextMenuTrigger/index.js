import { ContextMenuTrigger as Trigger } from 'react-contextmenu';
import React from 'react';
import { concat } from 'ramda';
import PropTypes from 'prop-types';

const ContextMenuTrigger = ({ id, type, ...props }) =>
  <Trigger id={concat(id, type)} {...props} />;

ContextMenuTrigger.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default ContextMenuTrigger;
