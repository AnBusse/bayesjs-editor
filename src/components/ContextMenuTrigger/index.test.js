import { shallow } from 'enzyme';
import React from 'react';
import ContextMenuTrigger from './index';

const defaultProps = {
  id: 'id',
  type: 'type',
  children: <div />,
};

const shallowComponent = () =>
  shallow(<ContextMenuTrigger {...defaultProps} />);

describe('ContextMenuTrigger Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });

  it('has id and type as prop id', () => {
    const component = shallowComponent();

    expect(component.prop('id')).toBe(`${defaultProps.id}${defaultProps.type}`);
  });
});
