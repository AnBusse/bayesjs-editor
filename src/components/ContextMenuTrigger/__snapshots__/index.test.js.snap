// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuTrigger Component matches snapshot 1`] = `
<ContextMenuTrigger
  attributes={Object {}}
  collect={[Function]}
  disable={false}
  disableIfShiftIsPressed={false}
  holdToDisplay={1000}
  id="idtype"
  posX={0}
  posY={0}
  renderTag="div"
>
  <div />
</ContextMenuTrigger>
`;
